tools/vendored.py

```diff
@@ -1,8 +1,10 @@
 import re
+import shutil
 import sys
 import subprocess
+from textwrap import dedent
 
-from path import Path
+from pathlib import Path
```

Comment:
This may break the script, right?

For example, in the `remove_all` function there is a call to `path.rmtree()` which should be available in `path.Path` but not `pathlib.Path`, I think.

(There might be other undefined methods also called, like `file.text()` in `rewrite_packaging`).

Reply:
If so, then I'm surprised that none of the checks raised any errors. I'll revise the patch. The reason I did this is because apparently the `.suffix` attribute was missing from `path.Path`, or at least both mypy and PyCharm complained abouit that.

Reply:
I think we are not running this script in any test, and since there is no type annotations, the checks also don't surface it.

Probably the only way to check if the script works well is by running `tox -e vendor`.

Reply:
Yes, there seems to be a problem there:

```
pipx run --python python3.12 tox -e vendor
⚠️  tox is already on your PATH and installed at /home/abravalheri/.local/bin/tox. Downloading and
    running anyway.
vendor: commands[0]> python -m tools.vendored
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 246, in <module>
    __name__ == '__main__' and update_vendored()
                               ^^^^^^^^^^^^^^^^^
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 16, in update_vendored
    update_pkg_resources()
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 198, in update_pkg_resources
    install(vendor)
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 181, in install
    clean(vendor)
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 177, in clean
    remove_all(path for path in vendor.glob('*') if path.basename() not in ignored)
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 11, in remove_all
    for path in paths:
  File "/home/abravalheri/workspace/setuptools/tools/vendored.py", line 177, in <genexpr>
    remove_all(path for path in vendor.glob('*') if path.basename() not in ignored)
                                                    ^^^^^^^^^^^^^
AttributeError: 'PosixPath' object has no attribute 'basename'
vendor: exit 1 (0.11 seconds) /home/abravalheri/workspace/setuptools> python -m tools.vendored pid=26031
  vendor: FAIL code 1 (0.13=setup[0.02]+cmd[0.11] seconds)
  evaluation failed :( (1.26 seconds)
```

Reply:
This particular problem can be easily fixed by using `.name` instead of `.basename`.

Reply:
> or at least both mypy and PyCharm complained abouit that.

Yeah, it seems to exist in the implementation or missing in the typestub:
https://github.com/jaraco/path/blob/2c42ff39cf1605a5c8ba039f48f451a3e24e0182/path/__init__.py#L327

Reply:
Maybe I should revert the import and mute the mypy error on that line.

Reply:
Done.